# Homework 1: Expressions  (50 points)

Please complete this notebook by filling in the cells provided. The solved homework should be uploaded to Canvas.

#### Reading:
- Data8 Textbook chapters [2](http://www.inferentialthinking.com/chapters/02/causality-and-experiments.html), 
 [3](http://www.inferentialthinking.com/chapters/03/programming-in-python.html) and 
[4](https://www.inferentialthinking.com/chapters/04/Data_Types).

#### Deadline:

This assignment is due on Tuesday, October 9 at 9:30am. Late work will not be accepted unless you have made special arrangements with the instructors.

Directly sharing answers is not okay, but discussing problems with the course staff or with other students is encouraged. 

You should start early so that you have time to get help if you're stuck. 

## 1. Scary Arithmetic (5 points)

An ad for ADT Security Systems says,

> "When you go on vacation, burglars go to work [...] According to FBI statistics, over 25% of home burglaries occur between Memorial Day and Labor Day."

Do the data in the ad support the claim that burglars are more likely to go to work during the time between Memorial Day and Labor Day than at other times? Please explain your answer.

The data does not support the claim. Although there is some percent of home burglaries during the time, there is no indication that there are significantly more burglaries during that time than other times. For example, 30% of burglaries may happen during the winter holidays. In addition, the time between Memorial Day and Labor Day is ~3 months, which is 25% of the year. If 25% of home burglaries happen over 25% of the year, then this may not be more than usual (although more data would be needed to support the assumption that the likelihood is constant as well).  Overall, 75% (which is 3x greater than 25%) of burglaries happen outside of that time, and therefore it is too vague to really draw any conclusions about the likelihood that burglaries are more likely to happen because there is not evidence on the other times of the year and more information.

## 2. Names and Assignment Statements (15 points)

**Question 1.** When you run the following cell, Python produces a cryptic error message.

In [2]:
4=2+2

SyntaxError: can't assign to literal (<ipython-input-2-765ef5e0d615>, line 1)

Choose the best explanation of what's wrong with the code, and then assign 1, 2, 3, or 4 to `names_q1` below to indicate your answer.

1. Python is smart and already knows `4 = 2 + 2`.

2. `4` is a number, and it doesn't make sense to make a number be a name for something else. In Python, "`x = 2 + 2`" means "assign `x` as the name for the value of `2 + 2`."

3. It should be `2 + 2 = 4`.

4. I don't get an error message. This is a trick question.

In [1]:
names_q1 = 2

Answer = 2

**Question 2.** When you run the following cell, Python will produce another cryptic error message.

In [4]:
two = 3
six = two plus two

SyntaxError: invalid syntax (<ipython-input-4-820d4d61e3dd>, line 2)

Choose the best explanation of what's wrong with the code and assign 1, 2, 3, or 4 to `names_q2` below to indicate your answer.

1. The `plus` operation only applies to numbers, not the word "two".

2. The name "two" cannot be assigned to the number 3.

3. Two plus two is four, not six.

4. The name `two` cannot be followed directly by another name.

In [3]:
names_q2 = 4

Answer = 4

**Question 3. Volume of a sphere** - finish the code in the following cell so that it calculates the volume of a sphere of radius 10, and it prints **"Volume of a sphere of radius 10 is ?"** (with "?" replace by the correct answer).

In [10]:
pi = 3.1415926535897931
r = 10
V = (4/3)*pi*(r**3)
print("Volume of a sphere of radius 10 is", V)

Volume of a sphere of radius 10 is 4188.790204786391


## 3. Colleges (30 points)

The question uses data that we donwloaded from Kaggle.com; more precisely from here:

https://www.kaggle.com/theriley106/university-statistics

The dataset was posted at the beginning of 2018, and was grabbed from US-News: https://www.usnews.com. It includes statistics for 311 US Universities. 

To make your tasks easiear, we removed many of the columns and kept only 11. The corresponding file can be found here:

http://www.stat.uchicago.edu/~nicolae/118/colleges1.csv

We will use these data in other homeworks. If you want, you can open it on your computer and get a better understanding of it. For example, there are universities for which there is incomplete data.

#### Run the next cell that contains code that will allow you to read the data into a table called "colleges_table". It also shows the first 10 rows.


In [13]:
from datascience import *

colleges_table=Table.read_table('http://www.stat.uchicago.edu/~nicolae/118/colleges1.csv')
colleges_table

act-avg | sat-avg | enrollment | acceptance-rate | rankingDisplayScore | percent-receiving-aid | cost-after-aid | tuition | displayName                           | overallRank | institutionalControl
32      | 1400    | 5400       | 7               | 100                 | 60                    | 16793          | 47140   | Princeton University                  | 1           | private
32      | 1430    | 6710       | 5               | 98                  | 55                    | 16338          | 48949   | Harvard University                    | 2           | private
32      | 1450    | 5941       | 8               | 96                  | 42                    | 27767          | 54825   | University of Chicago                 | 3           | private
32      | 1420    | 5472       | 6               | 96                  | 50                    | 18385          | 51400   | Yale University                       | 3           | private
32      | 1430    | 6113       | 6               | 95                  | 48                    | 21041          | 57208   | Columbia University                   | 5           | private
33      | 1460    | 4524       | 8               | 95                  | 58                    | 20331          | 49892   | Massachusetts Institute of Technology | 5           | private
31      | 1380    | 7034       | 5               | 95                  | 47                    | 19296          | 49617   | Stanford University                   | 5           | private
32      | 1380    | 10019      | 9               | 93                  | 47                    | 25441          | 53534   | University of Pennsylvania            | 8           | private
31      | 1380    | 6609       | 11              | 92                  | 41                    | 23374          | 53744   | Duke University                       | 9           | private
34      | 1510    | 979        | 8               | 91                  | 51                    | 23973          | 49908   | California Institute of Technology    | 10          | private
... (301 rows omitted)

# **Question 1.** 

**In the next cell, finish the code that has the goal of displaying the university with largest tuition (but only showing the university name and the tuition account).**

The first line of code uses 'select' to construct a table with only two columns.

The second line a code defines a variable called 'Tuition' by extracting the corrresponding column from our "colleges_table" table.

Run first the code as is and then modify it to achieve what it is asked above.

In [20]:
name_tuition=colleges_table.select('displayName','tuition')
Tuition=colleges_table.column('tuition')

colleges_table.where('tuition',max(Tuition))
name_tuition.where('tuition', max(Tuition))

displayName         | tuition
Columbia University | 57208

**Question 2.** 

You are now ready for a slightly more independent exercise. In the next cell, write code that displays for the universities that are ranked 14 the following: name, tuition, average of SAT scores. (in the order indicated)

Note that ranks can be found in the column called "overallRank". For example, in this data set, Columbia, MIT and Stanford are all ranked 5 because they have the same ranking score and there are 4 schools that have higher scores. There is no university ranked 4th. The code should be in a way that is trivial to modify to show this information for any rank, let's say rank 5.

In [23]:
# For universities ranked 14 display: name, tuition, average of SAT scores.  

name_tuition_SAT=colleges_table.select('displayName','tuition', 'sat-avg')
overallRank = colleges_table.column('overallRank')

rank = 14 ##change number to whatever rank you want to display

name_tuition_SAT.where(overallRank, rank)


displayName                           | tuition | sat-avg
Columbia University                   | 57208   | 1430
Massachusetts Institute of Technology | 49892   | 1460
Stanford University                   | 49617   | 1380

**Question 3.** 

**What is the average tuition rounded to the dollar for the universities in this dataset?** The output of the next cell should show the number that is the answer to this question. 

In [40]:
# Calculate the average tuition for the universities in this dataset (rounded)

avg_tuition = sum(colleges_table.column('tuition'))/len(colleges_table.column('tuition'))
int(round(avg_tuition))

31121


**Question 4.** 

**What is the average "cost-after-aid" rounded to the dollar for the universities ranked 14 in this dataset?** The output of the next cell should show the number that is the answer to this question. 



In [42]:
# Calculate the average cost-after-aid for universities ranked 14 (rounded)
name_cost=colleges_table.select('displayName','cost-after-aid')
overallRank = colleges_table.column('overallRank')

rank = 14 ##change number to whatever rank you want to display
rank_table = name_cost.where(overallRank, rank)

avg_tuition = sum(rank_table.column('cost-after-aid'))/len(rank_table.column('cost-after-aid'))
int(round(avg_tuition, 0))


24967

**Question 5.** 

Create a variable called 'Rounded_Tuition' that contains the tuition for these universities to the first two relevant digits (tuition of 54825 should be 54000 in the new variable). 

Display the name of all Universities that have rounded tuition equal to $45K.

In [69]:
# Display Universities with tuition between 45000 and 45999
name_tuition = colleges_table.select('displayName','tuition') #If you want JUST name, remove tuition from the parentheses
Tuition = colleges_table.column('tuition')//1000

name_tuition.where(Tuition, 45)

displayName                      | tuition
Rice University                  | 45608
Illinois Institute of Technology | 45864